ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/sync/DefaultSynchronizer.java

```diff
@@ -98,9 +116,25 @@ public DefaultSynchronizer(
                   worldStateDownloader));
     } else {
       this.fastSyncDownloader = Optional.empty();
+      this.stateQueueDirectory = null;
     }
   }
 
+  private void setupShutdownHook(final Path stateQueueDirectory) {
+    Runtime.getRuntime()
+        .addShutdownHook(
+            new Thread(
+                () -> {
+                  try {
+                    // Clean up this data for now (until fast sync resume functionality is in place)
+                    MoreFiles.deleteRecursively(
+                        stateQueueDirectory, RecursiveDeleteOption.ALLOW_INSECURE);
+                  } catch (IOException e) {
+                    LOG.error("Unable to clean up fast sync files: {}", stateQueueDirectory, e);
+                  }
+                }));
+  }
```

Comment:
It would probably be better to put this in a `stop()` method which is called from `Runner.close`. Pantheon already sets up a shutdown hook to call `Runner.close` and then when tests shutdown `Runner` without killing the whole JVM this clean up will run then and not be delayed.

Reply:
Also, I can't see where the Queue gets closed. It didn't matter with the in memory implementation but we do need to ensure that rocksdb is closed.  This may be a good place (though we'd need to ensure `WorldStateDownloader` is stopped and not going to try to access the queue anymore.

Reply:
Good feedback!  Pushed the cleanup into a `stop()` method here: https://github.com/PegaSysEng/pantheon/pull/746/commits/f3d4e817885923a522d5c36c99f085677e88fb71

Reply:
Yep, that all LGTM.  Merge away. :)